### ISSUE TITLE
The terminal process command 'C:\Program Files\PowerShell\7-preview\pwsh.exe' failed to launch (exit code: {2})

### ISSUE THREAD

**Issue by rahul0399** [microsoft/vscode#95985]
Issue Type: <b>Feature Request</b>

The terminal process command 'C:\WINDOWS\system32\cmd.exe' failed to launch (exit code: {2}) I've reinstalled it multiple times along with Salesforce CLI Integration & Salesforce Extension Pack. However, it is giving the same error again & again.

System details: Windows 10

I've already tried multiple solutions that are given in GitHub repository but it hasn't fixed yet.

looking for your prompt help to fix the issue...

VS Code version: Code 1.44.2 (ff915844119ce9485abfe8aa9076ec76b5300ddd, 2020-04-16T16:36:23.138Z)
OS version: Windows_NT x64 6.2.9200

**Comment by rahul0399** [comment]
Hello Tyiar> hope you are well there. Could you please help me to fix this issue on little priority, as my learning salesforce LWC and due to this issue can't proceed further in my learning path.

**Comment by rahul0399** [comment]
Hello Tyiar> hope you are well there. Could you please help me to fix this issue on little priority, as my learning for salesforce LWC has stopped due to this issue and can't proceed further on my learning path until this will not start working.

**Comment by Tyriar** [comment]
Does C:\Program Files\PowerShell\7-preview\pwsh.exe exist?

This error is normally due to some anti-virus software blocking the terminal from launching, see https://code.visualstudio.com/docs/editor/integrated-terminal#_how-do-i-fix-the-errors-connectnamedpipe-failed-windows-error-232-or-terminal-exited-with-code-x-on-windows

**Comment by rahul0399** [comment]
Yes, it is, I've downloaded this PowerShell 7 as my system was having PowerShell Version: 5.1.18362.752, so I thought maybe this older version of PowerShell creating trouble for this issue. 
after uninstalling this VS code app, I have downloaded this latest version of PowerShell and thereafter I have re-installed it but still, the issue is persisted. 

I was not sure how to exclude those files which are mentioned in that given link, so I have excluded the VS code file from the windows defender and checked with the issue but still, it didn't work. 

now I have deleted all the created projects and tried again to run the terminal. It is opening now but I can't write anything on that, just getting dark curser and nothing happening. 

<img width="522" alt="Capture1" src="https://user-images.githubusercontent.com/48702210/80247535-da580880-868b-11ea-856e-af26eabbaf55.PNG">

<img width="568" alt="SFDC" src="https://user-images.githubusercontent.com/48702210/80247558-e17f1680-868b-11ea-8e5a-31feb7f26bdf.PNG">

**Comment by rahul0399** [comment]
Now again, I've created a project of VSCodeQuickstart and when trying to open the terminal, it is giving me the same error of "The terminal process command 'C:\Program Files\PowerShell\7-preview\pwsh.exe' failed to launch (exit code: {2})"
<img width="765" alt="error" src="https://user-images.githubusercontent.com/48702210/80250113-cc58b680-8690-11ea-93fe-4ce9fbef2696.PNG">

**Comment by Tyriar** [comment]
Can you try trace logging when opening a terminal and see what it says? https://github.com/microsoft/vscode/wiki/Terminal-Issues#enabling-trace-logging

Note that it prints your terminal environment which may contain sensitive information.

**Comment by rahul0399** [comment]
Sure, PFA created logs.

<img width="960" alt="Capture__log1" src="https://user-images.githubusercontent.com/48702210/80285433-83613a80-8742-11ea-927c-fac17bc2f93b.PNG">
<img width="958" alt="Capture__log2" src="https://user-images.githubusercontent.com/48702210/80285436-88be8500-8742-11ea-9778-6825b60b1043.PNG">
<img width="959" alt="Capture__log3" src="https://user-images.githubusercontent.com/48702210/80285439-8c520c00-8742-11ea-891b-93b0f45cea69.PNG">
<img width="957" alt="Capture__log4" src="https://user-images.githubusercontent.com/48702210/80285441-907e2980-8742-11ea-9ea0-0383fd10ba2f.PNG">

**Comment by Tyriar** [comment]
All those file watcher errors don't look good 😕, but they're likely unrelated to the terminal problem.

I'm expecting logs to look like this:

```
[2020-04-27 07:12:30.771] [renderer2] [trace] CommandService#executeCommand workbench.action.terminal.new
[2020-04-27 07:12:30.772] [renderer2] [trace] telemetry/workbenchActionExecuted {"id":"workbench.action.terminal.new","from":"keybinding"}
[2020-04-27 07:12:30.772] [renderer2] [trace] terminalInstance#ctor (id: 2) {}
[2020-04-27 07:12:30.854] [renderer2] [trace] IPty#spawn C:\Program Files\PowerShell\7\pwsh.exe [] {"name":"pwsh.exe","cwd":"E:\\GitHub\\microsoft\\vscode","env":{snip},"cols":212,"rows":22,"useConpty":true,"conptyInheritCursor":false}
[2020-04-27 07:12:31.362] [renderer2] [debug] Terminal process ready (shellProcessId: 8092)
[2020-04-27 07:12:31.364] [renderer2] [trace] IPty#resize 212 22
```

**Comment by rahul0399** [comment]
Hi, Tyriar> Not sure what to do and how to check these logs. However, I have good news, that I was checking about this issue and tried whatever was coming to my mind. So, I uninstalled it from Revo Uninstaller and re-installed it back, and thereafter, I excluded their folder from the windows defender. And, Same I did with Salesforce CLI (pasted below 
<img width="405" alt="Capture_WD" src="https://user-images.githubusercontent.com/48702210/80393106-a9671600-88cd-11ea-81eb-29229001b39e.PNG">
screenshot for your reference)and you know what, it is working for me but not sure for how long it will work. Thanks for your help!
